### PR TITLE
422 introduce logged groups

### LINKED ISSUE #422: Introduce logged groups feature
## Impact 

Proxy log is very wordy.
In order to separate log notices from each other and make it possible to suppress some logging domains proxy needs to be extended with appropriate feature

[logged_group using reference](https://github.com/neonlabsorg/python-logged-groups)

### PR DESCRIPTION
**Pull request by rozhkovdmitrii**

### CODE REVIEW

**Review by vodolaz — DISMISSED** [review]

**Review by vasiliy-zaznobin — COMMENTED** [review]
In [proxy.log](https://buildkite.com/organizations/cyberway/pipelines/neon-proxy/builds/3433/jobs/ca7b9472-183a-4597-855e-340f7105ccfc/artifacts/28d431e1-e24b-4e41-bbd8-fd8f4b056d04) there are logs with different formats:
```
2022-01-17 18:16:44,763    DEBUG     62:Thread-10               SolanaProxyPlugin >>> 140671523895040 0x7ff0a4511400 {"jsonrpc": "2.0", "id": 134, "result": "0xd"} eth_getTransactionCount resp_time_ms= 2.183198928833008
2022-01-17 18:16:44,787 - pid:62 [I] access_log:226 - 172.31.0.1:47368 - POST /solana - 26.18 ms
2022-01-17 18:16:44,787    DEBUG     60:Thread-10               SolanaProxyPlugin <<< 140671523895040 0x7ff0a4381430 {"method":"eth_sendRawTransaction","params":
```

**Comment by vasiliy-zaznobin** [comment]
**62:Thread-10** and **60:Thread-10** are the same threads?

**Comment by rozhkovdmitrii** [comment]
> **62:Thread-10** and **60:Thread-10** are the same threads?

Different PID

**Comment by vasiliy-zaznobin** [comment]
the neon-сli output will also have to be able to produce output according to the log_cfg.json format?

**Comment by rozhkovdmitrii** [comment]
> the neon-сli output will also have to be able to produce output according to the log_cfg.json format?

That is good idea!

I don't think for now it will be implemented. As I know `neon-cli` logging format will be hard coded in the first approach. Let's talk about it inside a different issue

**Comment by vasiliy-zaznobin** [comment]
There is no logs from indexer.
[Now in develop](https://buildkite.com/cyberway/neon-proxy/builds/3441#7e1cba0b-c696-448d-b658-c6388e1a3cad) proxy.log there are  log strings from it:

```
2022-01-18 10:53:04,048 - pid:48 [D] gather_unknown_transactions:91 -   0 get_signatures_for_address 0
2022-01-18 10:53:04,049 - pid:48 [D] gather_unknown_transactions:95 - len(results) == 0
2022-01-18 10:53:04,049 - pid:48 [D] gather_unknown_transactions:118 - start getting receipts
2022-01-18 10:53:04,052 - pid:48 [D] gather_unknown_transactions:124 - (560, '4RR3MQyzpWqv7mSqgfgKdHRbeFBCPNJCXrc9hcawgQ4K5NEfh6vY9NxLM23UwduPhuAJpQmpME7f3MdAX5J978TP')
2022-01-18 10:53:04,052 - pid:48 [D] process_functions:656 - Start getting blocks
2022-01-18 10:53:04,070 - pid:48 [D] gather_blocks:731 - gather_blocks from 609 to 611
2022-01-18 10:53:04,077 - pid:48 [D] gather_blocks:737 - gather_blocks_ms: 24.966001510620117 last_height: 611 last_block_slot 611
2022-01-18 10:53:04,077 - pid:48 [D] process_functions:658 - Process receipts
2022-01-18 10:53:04,081 - pid:48 [D] process_receipts:698 - process_receipts_ms: 2.7801990509033203 transaction_receipts.len: 11 from 561 to 611 slots
2022-01-18 10:53:04,081 - pid:48 [D] process_functions:660 - remove not finalized data in range[609..611]
2022-01-18 10:53:04,083 - pid:48 [D] process_functions:662 - Unlock accounts
2022-01-18 10:53:05,085 - pid:48 [D] process_functions:75 - Start indexing
```

**Comment by rozhkovdmitrii** [comment]
> There is no logs from indexer. [Now in develop](https://buildkite.com/cyberway/neon-proxy/builds/3441#7e1cba0b-c696-448d-b658-c6388e1a3cad) proxy.log there are log strings from it:
> 
> ```
> 2022-01-18 10:53:04,048 - pid:48 [D] gather_unknown_transactions:91 -   0 get_signatures_for_address 0
> 2022-01-18 10:53:04,049 - pid:48 [D] gather_unknown_transactions:95 - len(results) == 0
> 2022-01-18 10:53:04,049 - pid:48 [D] gather_unknown_transactions:118 - start getting receipts
> 2022-01-18 10:53:04,052 - pid:48 [D] gather_unknown_transactions:124 - (560, '4RR3MQyzpWqv7mSqgfgKdHRbeFBCPNJCXrc9hcawgQ4K5NEfh6vY9NxLM23UwduPhuAJpQmpME7f3MdAX5J978TP')
> 2022-01-18 10:53:04,052 - pid:48 [D] process_functions:656 - Start getting blocks
> 2022-01-18 10:53:04,070 - pid:48 [D] gather_blocks:731 - gather_blocks from 609 to 611
> 2022-01-18 10:53:04,077 - pid:48 [D] gather_blocks:737 - gather_blocks_ms: 24.966001510620117 last_height: 611 last_block_slot 611
> 2022-01-18 10:53:04,077 - pid:48 [D] process_functions:658 - Process receipts
> 2022-01-18 10:53:04,081 - pid:48 [D] process_receipts:698 - process_receipts_ms: 2.7801990509033203 transaction_receipts.len: 11 from 561 to 611 slots
> 2022-01-18 10:53:04,081 - pid:48 [D] process_functions:660 - remove not finalized data in range[609..611]
> 2022-01-18 10:53:04,083 - pid:48 [D] process_functions:662 - Unlock accounts
> 2022-01-18 10:53:05,085 - pid:48 [D] process_functions:75 - Start indexing
> ```

Because those are suppressed by default

**Comment by vasiliy-zaznobin** [comment]
What should we do to enable logging from indexer?
What other logs are disabled by default?

**Comment by rozhkovdmitrii** [comment]
> What should we do to enable logging from indexer? What other logs are disabled by default?

image

**Comment by vasiliy-zaznobin** [comment]
Can this be changed on the fly? Or will you have to restart the proxy?

**Comment by rozhkovdmitrii** [comment]
> Can this be changed on the fly? Or will you have to restart the proxy?

It can't be changed on the fly from the box. The common way is to process SIGHUP

**Comment by vasiliy-zaznobin** [comment]
I think that by default all logs should be enabled on the DEBUG level.

**Comment by vasiliy-zaznobin** [comment]
I think that one more field needs to be added - this is the name of the group that was used when logging.

**Comment by vasiliy-zaznobin** [comment]
@mich-master please take into account that we added a new field (5-th) in a log string  - **group name**
Proxy
Indexer
...
So for neon-cli it should be an unique name 

```
2022-01-18 14:08:39,113    DEBUG     50:Thread-1             Proxy              SolanaProxyPlugin <<< 139999184443136 0x7f5419aa53a0 {"method":"eth_blockNumber","params":[],"id":93,"jsonrpc":"2.0"}
2022-01-18 14:08:39,113    DEBUG     50:Thread-1             Proxy              SolanaProxyPlugin Request payload: {'method': 'eth_blockNumber', 'params': [], 'id': 93, 'jsonrpc': '2.0'}
2022-01-18 14:08:39,114    DEBUG     50:Thread-1             Proxy                  EthereumModel eth_blockNumber 0x0
2022-01-18 14:08:39,114    DEBUG     50:Thread-1             Proxy              SolanaProxyPlugin >>> 139999184443136 0x7f5419aa53a0 {"jsonrpc": "2.0", "id": 93, "result": "0x0"} eth_blockNumber resp_time_ms= 0.5552768707275391
2022-01-18 14:08:39,114 - pid:50 [I] access_log:226 - 192.168.208.1:41582 - POST /solana - 60.94 ms
2022-01-18 14:08:39,195    DEBUG     48:MainThread         Indexer                        Indexer   1 get_signatures_for_address 1000
2022-01-18 14:08:39,284    DEBUG     48:MainThread         Indexer                        Indexer   2 get_signatures_for_address 1000
2022-01-18 14:08:39,368    DEBUG     48:MainThread         Indexer                        Indexer   3 get_signatures_for_address 1000
2022-01-18 14:08:39,460    DEBUG     48:MainThread         Indexer                        Indexer   4 get_signatures_for_address 1000
2022-01-18 14:08:39,543    DEBUG     48:MainThread         Indexer                        Indexer   5 get_signatures_for_address 796
2022-01-18 14:08:39,597    DEBUG     48:MainThread         Indexer                        Indexer   6 get_signatures_for_address 0
2022-01-18 14:08:39,597    DEBUG     48:MainThread         Indexer                        Indexer len(results) == 0
2022-01-18 14:08:39,597    DEBUG     48:MainThread         Indexer                        Indexer start getting receipts
2022-01-18 14:08:39,698    DEBUG     48:Thread-1           Indexer                        Indexer 100
2022-01-18 14:08:39,716    DEBUG     52:Thread-1             Proxy                     solana_cli Calling: solana --url http://solana:8899 config get
2022-01-18 14:08:39,751    DEBUG     52:Thread-1             Proxy                  EthereumModel worker id 1
2022-01-18 14:08:39,751    DEBUG     52:Thread-1             Proxy               neon_config_load loading the neon config dict for the first time!
2022-01-18 14:08:39,751    DEBUG     52:Thread-1             Proxy                       neon_cli Calling: neon-cli --commitment=recent --url http://solana:8899 --evm_loader=53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io -vvv neon-elf-params
2022-01-18 14:08:39,776    DEBUG     52:Thread-1             Proxy               neon_config_load {'NEON_TOKEN_MINT': 'HPsV9Deocecw3GeZv1FkAPNCBRfuVyfw9MMwjwRe1xaU', 'NEON_SEED_VERSION': '1', 'NEON_POOL_BASE': '4sW3SZDJB7qXUyCYKA7pFL8eCTfm3REr8oSiKkww7MaT', 'NEON_CHAIN_ID': '111', 'NEON_PAYMENT_TO_TREASURE': '5000', 'NEON_PAYMENT_TO_DEPOSIT': '5000', 'NEON_PKG_VERSION': '0.5.4-dev', 'NEON_TOKEN_MINT_DECIMALS': '9', 'NEON_REVISION': '189e69e083b3566d8d36757eb23905ea519ba1cf', 'NEON_ACCOUNT_MAX_SIZE': '256', 'load_time': 1642514919.776927, 'web3_clientVersion': 'Neon/v0.5.4-dev-189e69e083b3566d8d36757eb23905ea519ba1cf'}
```

**Comment by vasiliy-zaznobin** [comment]
Still found in the log of [the build](https://buildkite.com/cyberway/neon-proxy/builds/3443)
```
2022-01-18 14:08:39,114 - pid:50 [I] access_log:226 - 192.168.208.1:41582 - POST /solana - 60.94 ms
```

**Comment by vasiliy-zaznobin** [comment]
to filter by group, I suggest making group names unique, for example:
**Indexer_log_group**
**Proxy_log_group**
**Neon_cli_log_group**
or suggest your names

**Comment by otselnik** [comment]
> Still found in the log of [the build](https://buildkite.com/cyberway/neon-proxy/builds/3443)
> 
> ```
> 2022-01-18 14:08:39,114 - pid:50 [I] access_log:226 - 192.168.208.1:41582 - POST /solana - 60.94 ms
> ```

This log is from the proxy core(not our code). Not sure if we need to do anything with it.

**Comment by otselnik** [comment]
proxy/indexer/indexer_db.py
proxy/indexer/indexer.py (ReceiptsParserState)
proxy/indexer/utils.py (NeonTxInfo)

These files were changed recently and should be updated to use a new logger.

**Comment by vasiliy-zaznobin** [comment]
Let's add a single suffix to the name of the groups (a prefix is possible) and if we have lines in the logs without this suffix, then we do not log everything using groups. For example: **_log_group**

Then I will be able to set up such a filter in elastic, and we will be able to track it

**Comment by rozhkovdmitrii** [comment]
> I think that by default all logs should be enabled on the DEBUG level.

Done

**Comment by rozhkovdmitrii** [comment]
> > Still found in the log of [the build](https://buildkite.com/cyberway/neon-proxy/builds/3443)
> > ```
> > 2022-01-18 14:08:39,114 - pid:50 [I] access_log:226 - 192.168.208.1:41582 - POST /solana - 60.94 ms
> > ```
> 
> This log is from the proxy core(not our code). Not sure if we need to do anything with it.

We discussed it yesterday with @vodolaz and decided do nothing with it

**Comment by rozhkovdmitrii** [comment]
> I think that one more field needs to be added - this is the name of the group that was used when logging.

Done

**Comment by rozhkovdmitrii** [comment]
> Still found in the log of [the build](https://buildkite.com/cyberway/neon-proxy/builds/3443)
> 
> ```
> 2022-01-18 14:08:39,114 - pid:50 [I] access_log:226 - 192.168.208.1:41582 - POST /solana - 60.94 ms
> ```

That logs will be ignored

**Review by vasiliy-zaznobin — DISMISSED** [review]

File: proxy/indexer/pythnetwork.py [hunk 1]
```diff
@@ -48,21 +46,20 @@ def unpack(layout_descriptor, raw_data, field_name, index = 0):
         return read_dict(raw_data[start_idx:stop_idx])
     return struct.unpack(field['format'], raw_data[start_idx:stop_idx])[0]
 
-  
+
+@logged_group("Indexer")
 class PythNetworkClient:
```
Comment (vasiliy-zaznobin) [inline proxy/indexer/pythnetwork.py]: I would suggest a separate group for the airdropper

Comment (rozhkovdmitrii) [reply proxy/indexer/pythnetwork.py]: Yeah, I see. Thank you !

Comment (rozhkovdmitrii) [reply proxy/indexer/pythnetwork.py]: Done.

**Review by vasiliy-zaznobin — DISMISSED** [review]

**Review by otselnik — DISMISSED** [review]

**Review by afalaleev — DISMISSED** [review]

**Review by afalaleev — DISMISSED** [review]

**Review by afalaleev — APPROVED** [review]

**Pull request merged**